app/controllers/users_controller.rb

```diff
@@ -18,14 +18,26 @@ def new
   def create
     @user = User.new(user_params)
     if @user.save
-      session[:user_id] = @user.id
-      flash[:notice] = "Welcome #{@user.username}! You have successfully signed up."
-      redirect_to articles_path
+      UserMailer.welcome_email(@user).deliver_now
+      flash[:notice] = "Please check your emails to confirm your sign up"
+      redirect_to root_path
     else
+      flash[:error] = "Ooooppss, something went wrong!"
       render 'new'
     end
   end
 
+  def confirm_email
+    user = User.find_by_confirm_token(params[:id])
```

Comment:
Warum heißt der Parameter `id`? Ich hätte erwartet, dass es

```suggestion
    user = User.find_by_confirm_token(params[:token])
```
oder ähnlich heißt?